mockserver/data_reader.go

```diff
@@ -28,8 +29,37 @@ type BlockHeader struct {
 	NextBlock chainhash.Hash `json:"nextblockhash"`
 }
 
+type Transaction struct {
```

Comment:
Is this a copy? The transaction type should be already defined in the btcd module - so no need to copy

Reply:
In btcd, there are atleast 2 structs, one internal for p2p, which are in `wire` module, like `wire.Tx` and there is another for returning as RPC response, those are in `btcjson` module. 

https://github.com/btcsuite/btcd/blob/master/wire/msgtx.go#L364-L369
https://github.com/btcsuite/btcd/blob/master/btcjson/walletsvrresults.go#L146-L159

In case of txn, I've used `btcjson` one if the whole struct is returned, and if it is used internally then the same `wire.Tx` would require modifications when returning over RPC, hence this way is simpler and easier to read.

Reply:
Please add a comment to the Tx type, with a link to the code or spec where the the Bitcoin transaction type is defined (ideally in btcd). The links you provided above are not related to this one. 
BTW, We need a block tx type, not a transaction result.

Reply:
The custom types are now removed. Only using `btcjson` ones.